File: problems/js/twoSum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
  var found = {};
  var j;

  for (var i = 0; i < nums.length; i++) {
      if (!found[nums[i]]) {
          found[nums[i]] = i;
      }
  }

  for (var i = 0; i < nums.length; i++) {
      j = target - nums[i];
      if (found[j] && found[j] !== i) {
          return [i, found[j]];
      }
  }
};

File: problems/python/twoSum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:

        # Iterate over list, and store values as we go
        seenSoFar = {}


        # Iterate
        for index, num in enumerate(nums):

            # Check if we've found answer
            diff = target - num
            diffIndex = seenSoFar.get(diff)

            if diffIndex != None:
                return [diffIndex, index]


            # If not, add to seen and keep going
            seenSoFar[num] = index


        # If no answer, return []
        return []
